### PR TITLE
Added new play create api

### PR DESCRIPTION
**Pull request by appu-channelape**

### CODE REVIEW

File: src/plays/service/PlaysService.ts [hunk 1]
```diff
@@ -27,6 +32,15 @@ export default class PlaysService {
     });
     return deferred.promise as any;
   }
+  public create(play: PlayCreateRequest): Promise<Play> {
+    const deferred = Q.defer<Play | Play[]>();
+    const requestUrl = `/${Version.V2}${Resource.PLAYS}`;
+    this.client.put(requestUrl, { data: play }, (error, response, body) => {
```
Comment (ryankazokas) [inline src/plays/service/PlaysService.ts:38]: @appu-channelape this should be POST, not PUT. Please fix it to be the correct method for creating plays

**Review by ryankazokas — CHANGES_REQUESTED** [review]
Just some small feedback here. Please let me know once you are done and I'll re-review. Thanks @appu-channelape !

File: src/plays/service/PlaysService.ts (hunk 1 above)
Comment (appu-channelape) [reply src/plays/service/PlaysService.ts:38]: oh good catch @ryankazokas thank you

File: test/plays/service/PlaysService.spec.ts [hunk 2]
```diff
@@ -212,6 +212,25 @@ describe('Plays Service', () => {
...
+    it('And valid create play data ' +
+      'When creating play Then return resolved promise with play', () => {
+      const response = {
+        status: 200,
+        config: {
+          method: 'POST'
+        }
+      };
+      const clientGetStub: sinon.SinonStub = sandbox.stub(client, 'post')
+        .yields(null, response, expectedPlay);
```
[the remark this answers is not in the document]
Comment (appu-channelape) [reply test/plays/service/PlaysService.spec.ts:224]: please correct me If I am wrong. Do you mean like `const clientGetStub: sinon.SinonStub = sandbox
        .stub(client, "post")
        .yields(null, response, {
          id: "9c728601-0286-457d-b0d6-ec19292d4485",
          name: "Create new Play",
          createdAt: new Date("2018-02-22T16:04:29.030Z"),
          updatedAt: new Date("2018-04-02T13:04:27.299Z"),
          scheduleConfigurations: [
            {
              id: 1,
              targetAction: "INVOKE_PLAY",
              daysOfWeek: ["MONDAY", "TUESDAY", "WEDNESDAY"],
              endTimeInMinutes: 1440,
              intervalTimeInMinutes: 1337,
              startTimeInMinutes: 544,
              createdAt: new Date("2018-02-22T16:04:29.030Z"),
              updatedAt: new Date("2018-04-02T13:04:27.299Z"),
            }
          ]
        });`

[the remark this answers is not in the document]
Comment (ryankazokas) [reply test/plays/service/PlaysService.spec.ts:224]: Sorry, you are good here @appu-channelape . Please ignore the above comment. I didn't realize the third param was the response body. This looks good.

File: src/plays/service/PlaysService.ts (hunk 1 above)
Comment (mb-ca) [reply src/plays/service/PlaysService.ts]: To catch small things like this before pushing, you can E2E test by creating a simple local JS file and then installing your local channelape-sdk using `npm link`.

I have a local "sandbox" folder with a package.json and index.js file with the following which I use to test my SDK changes:
```
const channelApeSdk = require('channelape-sdk');
const client = new channelApeSdk.ChannelApeClient({
	minimumRequestRetryRandomDelay: 1000,
	maximumRequestRetryRandomDelay: 3000,
	maximumRequestRetryTimeout: 3000,
	sessionId: SESSION_ID,
	maximumConcurrentConnections: 10
});
run();
async function run() {
	try {
		const res = await client.plays().get();
		console.log(res);
	} catch (error) {
		console.log(error);
	}
	console.log('done');
}
```

I call `node index.js` to run it.

Some instructions I pulled and modified from our web app repo regarding using `npm link`:
To use changes you have made in the your local channelape-sdk repository before they are published to NPM, follow these steps:

1. In the channelape-sdk:
    1. Run `npm run compile` (important - must be done after every change you make to the sdk repo)
    2. Run `npm link`
2. In the repo you wish to use your local channelape-sdk version:
    1. Run `npm link channelape-sdk`
    
If you need to stop referencing the local channelape-sdk version, run `npm unlink channelape-sdk`.

File: src/index.ts [hunk 3]
```diff
@@ -104,6 +104,7 @@ export { default as PlaySettings } from './suppliers/model/PlaySettings';
 export { default as Play } from './plays/model/Play';
 export { default as PlayScheduleConfiguration } from './plays/model/PlayScheduleConfiguration';
 export { default as PlayUpdateRequest } from './plays/model/PlayUpdateRequest';
+export { default as PlayCreateRequest } from "./plays/model/PlayCreateRequest";
```
Comment (ryankazokas) [inline src/index.ts:107]: @appu-channelape Can you change this single quote like the build server is suggesting?

File: src/plays/model/PlayCreateRequest.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import PlayScheduleConfiguration from "./PlayScheduleConfiguration";
```
Comment (ryankazokas) [inline src/plays/model/PlayCreateRequest.ts:1]: @appu-channelape Change this to the single quotes like the build server is asking

**Review by ryankazokas — APPROVED** [review]
@appu-channelape get the build passing and this looks good to merge. Thanks for handling this.

**Pull request merged**